### PR TITLE
Middleware

### PR DESCRIPTION
**Pull request by maurafortino**
What's Included:
- added otel and candlelight middleware for core endpoints
- created a Handler Module
- updated metrics in http.go from gokit to prometheus

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -125,6 +129,7 @@ func ProvideMetrics() fx.Option {
 			Name: DropsDueToPanic,
 			Help: "The outgoing message delivery pipeline panicked.",
 		}, UrlLabel),
+
```
Comment (denopink) [inline metrics.go:132]:
```suggestion
```

File: routes.go [hunk 2]
```diff
@@ -53,8 +61,23 @@ func provideCoreEndpoints() fx.Option {
 func provideCoreOption(server string, in RoutesIn) arrangehttp.Option[http.Server] {
 	return arrangehttp.AsOption[http.Server](
 		func(s *http.Server) {
+			urlPrefix := fmt.Sprintf("/%s", apiBase)
+			if in.PreviousVersionSupport {
+				urlPrefix = fmt.Sprintf("/%s", apiBaseDualVersion)
+			}
 			mux := chi.NewMux()
+			// TODO: should probably customize things a bit
+			mux.Use(recovery.Middleware(recovery.WithStatusCode(555)))
```
Comment (denopink) [inline routes.go:70]: is this a later todo?

File: routes.go [hunk 3]
```diff
@@ -53,8 +61,23 @@ func provideCoreEndpoints() fx.Option {
 func provideCoreOption(server string, in RoutesIn) arrangehttp.Option[http.Server] {
 	return arrangehttp.AsOption[http.Server](
 		func(s *http.Server) {
+			urlPrefix := fmt.Sprintf("/%s", apiBase)
+			if in.PreviousVersionSupport {
+				urlPrefix = fmt.Sprintf("/%s", apiBaseDualVersion)
+			}
 			mux := chi.NewMux()
```
Comment (denopink) [inline routes.go:68]:
```suggestion
			}
			
			mux := chi.NewMux()
```

File: routes.go [hunk 4]
```diff
@@ -53,8 +61,23 @@ func provideCoreEndpoints() fx.Option {
 func provideCoreOption(server string, in RoutesIn) arrangehttp.Option[http.Server] {
 	return arrangehttp.AsOption[http.Server](
 		func(s *http.Server) {
+			urlPrefix := fmt.Sprintf("/%s", apiBase)
+			if in.PreviousVersionSupport {
+				urlPrefix = fmt.Sprintf("/%s", apiBaseDualVersion)
+			}
```
Comment (denopink) [inline routes.go:67]:
```suggestion
			}
			
```

Comment (denopink) [inline routes.go:67]:
```suggestion
			}
			
```

**Review by denopink — APPROVED** [review]
besides `httpClient_test.go` and `http_test.go` being completely commented out (assuming we're gonna those back in the near future or in this pr), lgtm

File: routes.go (hunk 2 above)
Comment (maurafortino) [reply routes.go:71]: this was copied and pasted over from old code - including the comment.

**Comment by maurafortino** [comment]
> besides `httpClient_test.go` and `http_test.go` being completely commented out (assuming we're gonna those back in the near future or in this pr), lgtm

yeah had to comment out the test files because they were using functions that were also being commented out due to ancla/argus issue. will comment back in once we get that fixed.

**Pull request merged**